domain/timeline/src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/domain/timeline/TimelineEvent.kt

```diff
@@ -1,15 +1,82 @@
 package uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.timeline
 
 import java.time.Instant
+import java.util.UUID
 
 /**
- * A TimelineEvent represents a single event that occurred in the system or potentially the wider prison estate.
+ * A TimelineEvent represents a single event that occurred whilst a prisoner has been in prison.
  *
  * Once created a TimelineEvent cannot be mutated in any way. It represents a single event at a point in time.
  */
 data class TimelineEvent(
-  val title: String,
-  val summary: String,
-  val eventDateTime: Instant,
-  val links: List<Link> = emptyList(),
-)
+  /**
+   * A unique reference for each event.
+   */
+  val reference: UUID,
+  /**
+   * The reference of the entity that's being updated, such as a Goal or an individual Goal's Step. A [String] rather
+   * than a [UUID], because we cannot guarantee the format from other systems (e.g. the ciag-induction-service).
+   */
+  val sourceReference: String,
+  /**
+   * The type of the event - see [TimelineEventType].
+   */
+  val eventType: TimelineEventType,
+  /**
+   * Some useful contextual information about the event. For example a reason for the event.
+   */
+  val contextualInfo: String?,
```

Comment:
I'm not sure about adding this field (if at all). I might remove it as I don't think we'll use it for MVP.

Reply:
Agreed to keep it as I think we'll need it at some stage and it will save flyway scripts etc later.